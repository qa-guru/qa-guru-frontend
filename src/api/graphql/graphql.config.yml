schema: schema.graphql
extensions:
  endpoints:
    Default GraphQL Endpoint:
      url: ${VITE_APP_ENDPOINT}${VITE_GRAPHQL_URI}
      headers:
        user-agent: JS GraphQL
      introspect: false
