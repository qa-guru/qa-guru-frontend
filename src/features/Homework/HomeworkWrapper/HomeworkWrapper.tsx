import React, { useState } from "react";
import HomeworkAnswer from "../Homework/HomeworkContainer";
import UpdateHomework from "../UpdateHomework/UpdateHomeworkContainer";
import SendHomeWorkToCheck from "../SendHomeWork/SendHomeWorkContainer";
import { Button, Paper } from "@mui/material";
import { IHomeworkWrapper } from "./HomeworkWrapper.types";

const style = {
  paper: { padding: "20px", mt: "40px" },
  buttonUpdate: { textTransform: "none", minWidth: "151px" },
};

const HomeworkWrapper: React.FC<IHomeworkWrapper> = ({
  dataHomeworkStatus,
}) => {
  const [openHomeWorkEdit, setOpenHomeWorkEdit] = useState<Boolean>(false);

  let homework;

  if (
    dataHomeworkStatus?.homeWorkByStudentAndLecture?.status &&
    !openHomeWorkEdit
  ) {
    homework = <HomeworkAnswer />;
  } else if (
    dataHomeworkStatus?.homeWorkByStudentAndLecture?.status &&
    openHomeWorkEdit
  ) {
    homework = <UpdateHomework setOpenHomeWorkEdit={setOpenHomeWorkEdit} />;
  } else if (
    !dataHomeworkStatus?.homeWorkByStudentAndLecture?.status &&
    !openHomeWorkEdit
  ) {
    homework = <SendHomeWorkToCheck />;
  }
  return (
    <Paper sx={style.paper}>
      {homework}
      <br />
      {!openHomeWorkEdit &&
        dataHomeworkStatus?.homeWorkByStudentAndLecture?.status && (
          <Button
            onClick={() => setOpenHomeWorkEdit(true)}
            sx={style.buttonUpdate}
            variant="contained"
          >
            Редактировать
          </Button>
        )}
    </Paper>
  );
};

export default HomeworkWrapper;
