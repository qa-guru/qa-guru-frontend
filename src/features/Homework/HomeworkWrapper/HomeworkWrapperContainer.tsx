import React from "react";
import { useHomeWorkByStudentAndLectureStatusQuery } from "../../../api/graphql/homework/homeWorkByStudentAndLectureStatus";
import { useParams } from "react-router-dom";
import { useUserIdQuery } from "../../../api/graphql/user/userId";
import HomeworkWrapper from "./HomeworkWrapper";

const HomeworkWrapperContainer: React.FC = () => {
  const { lectureId } = useParams();
  const { data: dataUserId } = useUserIdQuery();
  const { data: dataHomeworkStatus } =
    useHomeWorkByStudentAndLectureStatusQuery({
      variables: { studentId: dataUserId?.user?.id!, lectureId: lectureId! },
    });

  return <HomeworkWrapper dataHomeworkStatus={dataHomeworkStatus!} />;
};

export default HomeworkWrapperContainer;
