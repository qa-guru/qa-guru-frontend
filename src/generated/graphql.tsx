import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  BigDecimal: any;
  BigInteger: any;
  Date: any;
  DateTime: any;
  LocalDateTime: any;
  LocalTime: any;
  Long: any;
  Time: any;
  Timestamp: any;
  Url: any;
  Void: any;
};

export type ContentFileDto = {
  __typename?: 'ContentFileDto';
  fileLocation?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
  size?: Maybe<Scalars['Long']>;
  type?: Maybe<Scalars['String']>;
};

export type HomeWorkDto = {
  __typename?: 'HomeWorkDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  subject?: Maybe<Scalars['String']>;
};

export type HomeWorkInput = {
  description?: InputMaybe<Scalars['String']>;
  id?: InputMaybe<Scalars['ID']>;
  subject?: InputMaybe<Scalars['String']>;
};

export type HomeWorkSort = {
  field?: InputMaybe<HomeWorkSortField>;
  order?: InputMaybe<Order>;
};

export enum HomeWorkSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type HomeWorksDto = {
  __typename?: 'HomeWorksDto';
  items?: Maybe<Array<Maybe<HomeWorkDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

/** Mutation root */
export type Mutation = {
  __typename?: 'Mutation';
  /** purchase section */
  createPurchase?: Maybe<PurchaseDto>;
  /** training section */
  createTraining?: Maybe<TrainingDto>;
  /** user section */
  createUser?: Maybe<UserDto>;
  deleteHomeWork?: Maybe<Scalars['Void']>;
  deleteTask?: Maybe<Scalars['Void']>;
  deleteUser?: Maybe<Scalars['Void']>;
  /** homeWork section */
  updateHomeWork?: Maybe<HomeWorkDto>;
  /** person section */
  updatePerson?: Maybe<PersonDto>;
  updateRole?: Maybe<UserDto>;
  /** task section */
  updateTask?: Maybe<TaskDto>;
};


/** Mutation root */
export type MutationCreatePurchaseArgs = {
  input: PurchaseInput;
};


/** Mutation root */
export type MutationCreateTrainingArgs = {
  input: TrainingInput;
};


/** Mutation root */
export type MutationCreateUserArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


/** Mutation root */
export type MutationDeleteHomeWorkArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteTaskArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteUserArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationUpdateHomeWorkArgs = {
  input: HomeWorkInput;
};


/** Mutation root */
export type MutationUpdatePersonArgs = {
  input: PersonInput;
};


/** Mutation root */
export type MutationUpdateRoleArgs = {
  id: Scalars['ID'];
  roles?: InputMaybe<Array<InputMaybe<UserRole>>>;
};


/** Mutation root */
export type MutationUpdateTaskArgs = {
  input: TaskInput;
};

export enum Order {
  Asc = 'ASC',
  Desc = 'DESC'
}

export type PersonDto = {
  __typename?: 'PersonDto';
  avatarLocation?: Maybe<Scalars['String']>;
  firstName?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  lastName?: Maybe<Scalars['String']>;
  middleName?: Maybe<Scalars['String']>;
  phoneNumber?: Maybe<Scalars['String']>;
};

export type PersonInput = {
  firstName: Scalars['String'];
  lastName: Scalars['String'];
  middleName?: InputMaybe<Scalars['String']>;
  phoneNumber: Scalars['String'];
};

export type PurchaseDto = {
  __typename?: 'PurchaseDto';
  id?: Maybe<Scalars['ID']>;
  training: TrainingDto;
  trainingTariff: TrainingTariffDto;
  user: UserDto;
};

export type PurchaseInput = {
  trainingId?: InputMaybe<Scalars['ID']>;
  trainingTariffId?: InputMaybe<Scalars['ID']>;
  userId?: InputMaybe<Scalars['ID']>;
};

/** Query root */
export type Query = {
  __typename?: 'Query';
  /** homeWork section */
  homeWork?: Maybe<HomeWorkDto>;
  homeWorks?: Maybe<HomeWorksDto>;
  /** person section */
  person?: Maybe<PersonDto>;
  personByUserId?: Maybe<PersonDto>;
  /** task section */
  task?: Maybe<TaskDto>;
  taskHomeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  tasks?: Maybe<TasksDto>;
  /** user section */
  user?: Maybe<UserDto>;
  users?: Maybe<UsersDto>;
};


/** Query root */
export type QueryHomeWorkArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryHomeWorksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<HomeWorkSort>;
};


/** Query root */
export type QueryPersonByUserIdArgs = {
  id: Scalars['ID'];
};


/** Query root */
export type QueryTaskArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTaskHomeWorksArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTasksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<TaskSort>;
};


/** Query root */
export type QueryUserArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryUsersArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<UserSort>;
};

export type TaskDto = {
  __typename?: 'TaskDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  homeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  id?: Maybe<Scalars['ID']>;
  last?: Maybe<Scalars['ID']>;
  locking?: Maybe<Scalars['Boolean']>;
  speakers?: Maybe<Array<Maybe<UserInfoDto>>>;
  subject?: Maybe<Scalars['String']>;
};

export type TaskInput = {
  description?: InputMaybe<Scalars['String']>;
  homeWorks?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  id?: InputMaybe<Scalars['ID']>;
  last?: InputMaybe<Scalars['ID']>;
  locking?: InputMaybe<Scalars['Boolean']>;
  speakers?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  subject?: InputMaybe<Scalars['String']>;
};

export type TaskSort = {
  field?: InputMaybe<TaskSortField>;
  order?: InputMaybe<Order>;
};

export enum TaskSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type TasksDto = {
  __typename?: 'TasksDto';
  items?: Maybe<Array<Maybe<TaskDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export enum TechStack {
  Java = 'JAVA',
  Python = 'PYTHON'
}

export type TrainingDto = {
  __typename?: 'TrainingDto';
  id: Scalars['ID'];
  name: Scalars['String'];
  tariffs?: Maybe<Array<Maybe<TrainingTariffDto>>>;
  techStack: TechStack;
};

export type TrainingInput = {
  name?: InputMaybe<Scalars['String']>;
  tariffs?: InputMaybe<Array<InputMaybe<TrainingTariffInput>>>;
  techStack: TechStack;
};

export type TrainingTariffDto = {
  __typename?: 'TrainingTariffDto';
  code?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
};

export type TrainingTariffInput = {
  code?: InputMaybe<Scalars['String']>;
  description?: InputMaybe<Scalars['String']>;
  name?: InputMaybe<Scalars['String']>;
};

export type UserDto = {
  __typename?: 'UserDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export type UserIdInput = {
  id?: InputMaybe<Scalars['ID']>;
};

export type UserInfoDto = {
  __typename?: 'UserInfoDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  person?: Maybe<PersonDto>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export enum UserRole {
  Admin = 'ADMIN',
  Lector = 'LECTOR',
  Manager = 'MANAGER',
  Master = 'MASTER',
  Mentor = 'MENTOR',
  User = 'USER'
}

export type UserSort = {
  field?: InputMaybe<UserSortField>;
  order?: InputMaybe<Order>;
};

export enum UserSortField {
  Email = 'EMAIL',
  LastName = 'LAST_NAME',
  Phone = 'PHONE'
}

export type UsersDto = {
  __typename?: 'UsersDto';
  items?: Maybe<Array<Maybe<UserInfoDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export type SignUpMutationVariables = Exact<{
  username: Scalars['String'];
  password: Scalars['String'];
}>;


export type SignUpMutation = { __typename?: 'Mutation', createUser?: { __typename?: 'UserDto', id?: string | null, email?: string | null, roles?: Array<UserRole | null> | null } | null };


export const SignUpDocument = gql`
    mutation SignUp($username: String!, $password: String!) {
  createUser(username: $username, password: $password) {
    id
    email
    roles
  }
}
    `;
export type SignUpMutationFn = Apollo.MutationFunction<SignUpMutation, SignUpMutationVariables>;

/**
 * __useSignUpMutation__
 *
 * To run a mutation, you first call `useSignUpMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useSignUpMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [signUpMutation, { data, loading, error }] = useSignUpMutation({
 *   variables: {
 *      username: // value for 'username'
 *      password: // value for 'password'
 *   },
 * });
 */
export function useSignUpMutation(baseOptions?: Apollo.MutationHookOptions<SignUpMutation, SignUpMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<SignUpMutation, SignUpMutationVariables>(SignUpDocument, options);
      }
export type SignUpMutationHookResult = ReturnType<typeof useSignUpMutation>;
export type SignUpMutationResult = Apollo.MutationResult<SignUpMutation>;
export type SignUpMutationOptions = Apollo.BaseMutationOptions<SignUpMutation, SignUpMutationVariables>;