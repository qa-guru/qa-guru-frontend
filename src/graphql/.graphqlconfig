{
  "name": "Untitled GraphQL Schema",
  "schemaPath": "schema.graphql",
  "extensions": {
    "endpoints": {
      "Default GraphQL Endpoint": {
        "url": "http://5.161.103.109:8080/graphql",
        "headers": {
          "user-agent": "JS GraphQL"
        },
        "introspect": false
      }
    }
  }
}