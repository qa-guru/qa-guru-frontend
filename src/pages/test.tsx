import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
