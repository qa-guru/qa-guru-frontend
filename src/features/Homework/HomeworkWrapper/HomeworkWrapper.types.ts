import { HomeWorkByStudentAndLectureQuery } from "../../../generated/graphql";

export interface IHomework {
  data: HomeWorkByStudentAndLectureQuery;
}
