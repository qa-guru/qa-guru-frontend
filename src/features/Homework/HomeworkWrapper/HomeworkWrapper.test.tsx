import { MemoryRouter } from "react-router-dom";
import { render } from "../../../test/utilsTest";
import HomeworkWrapper from "./HomeworkWrapper";
import { HomeWorkByStudentAndLectureQuery } from "../../../generated/graphql";

const data: HomeWorkByStudentAndLectureQuery = {
  __typename: "Query",
  homeWorkByStudentAndLecture: {
    __typename: "StudentHomeWorkDto",
    id: "string",
    answer: "string",
    status: null,
    creationDate: new Date(),
    startCheckingDate: new Date(),
    endCheckingDate: new Date(),
    lecture: {
      __typename: "LectureInfoDto",
      id: "string",
      subject: "string",
    },
    student: {
      __typename: "UserDto",
      id: "string",
      firstName: "string",
      middleName: "string",
      lastName: "string",
    },
    mentor: {
      __typename: "UserDto",
      id: "string",
      firstName: "string",
      middleName: "string",
      lastName: "string",
    },
  },
};

describe("HomeworkWrapper", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomeworkWrapper data={data!} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
