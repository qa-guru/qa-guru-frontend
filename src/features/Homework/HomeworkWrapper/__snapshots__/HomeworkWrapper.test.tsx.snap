// Vitest Snapshot v1

exports[`HomeworkWrapper > the component is render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w6a372-MuiPaper-root"
  >
    <div
      class="css-a2l1ks-MuiStack-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Ответ на задание
      </h5>
      <div
        class="css-zpn3py-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          />
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-tr0r3x"
    >
      <form>
        <div
          class="css-17qspa5-MuiStack-root"
        >
          <div
            class="MuiBox-root css-8atqhb"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  placeholder="Текст ответа"
                  rows="5"
                  style="height: 0px; overflow: hidden;"
                />
                <textarea
                  aria-hidden="true"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                  tabindex="-1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1tn15vr-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              id=":r1:"
              tabindex="0"
              type="button"
            >
              Отправить
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
